File: 033_Laplacian_Filter/033_Laplacian_Filter.py
import cv2
import numpy as np

image = cv2.imread("H:\\OpenCV_Learning\\test.jpg")
cv2.namedWindow("input", cv2.WINDOW_AUTOSIZE)
cv2.imshow("input", image)

h, w = image.shape[:2]
src = cv2.GaussianBlur(image, (3, 3), 1)
dst = cv2.Laplacian(src, cv2.CV_32F, ksize=3, delta=127)
dst = cv2.convertScaleAbs(dst)

result = np.zeros([h, w*2, 3], dtype=image.dtype)
result[0:h, 0:w, :] = image
result[0:h, w:w*2, :] = dst
cv2.imshow("result", result)
cv2.imwrite("laplacian.png", result)

cv2.waitKey(0)
cv2.destroyAllWindows()

File: 033_Laplacian_Filter/033_Laplacian_Filter/033_Laplacian_Filter.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

using namespace std;
using namespace cv;

int main(int argc, char** argv) {
	Mat src = imread("H:\\OpenCV_Learning\\test.jpg");
	if (src.empty()) {
		printf("could not load img...\n");
		return -1;
	}
	namedWindow("input", WINDOW_AUTOSIZE);
	imshow("input", src);

	Mat blured, dst;
	GaussianBlur(src, blured, Size(3, 3), 0);
	// 为什么要设置127.0：二阶导求出来的差异较小，直接输出会是一片漆黑
	// 目的：提高亮度 原图 + 127
	Laplacian(blured, dst, CV_32F, 1, 1.0, 127.0, BORDER_DEFAULT);
	convertScaleAbs(dst, dst);
	imshow("result", dst);

	waitKey(0);
	destroyAllWindows();
	return 0;
}